co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java

```diff
@@ -211,101 +290,76 @@ public void initialize(long startTime, long endTime) throws InternalFederateExce
-        {
-            throw new InternalFederateException("Carla Ambassador initialization failed due to CARLA CDA Sim Adapter" 
-                + "connection! Check carla_config.json!", m);
-        }
         
-    }
-
-    /**
-     * Connects to CARLA simulator using the given host and port.
-     *
-     * @param host host on which CARLA simulator is running.
-     * @param port port on which CARLA client is listening.
-     */
-    @Override
-    public void connectToFederate(String host, int port) {
-        // Start the Carla connection server
-        String bridgePath = null;
-        int carlaConnectionPort = 8913;
+        // Load specified map if configured
         
-        if (carlaConfig.carlaConnectionPort != 0)
-            carlaConnectionPort = carlaConfig.carlaConnectionPort; // set the carla connection port
-
-        // get the connection bridge file
-        if (carlaConfig.bridgePath != null) {
-            bridgePath = carlaConfig.bridgePath;
-            log.info("Use connection bridge path from configuration file: " + carlaConfig.bridgePath);
-        } else {
-            log.error("Could not find connection bridge.");
-            return;
-        }
-        if (carlaConnection == null) {
-            // start the carla connection
-
-            carlaConnection = new CarlaConnection("localhost", carlaConnectionPort, this);
-            Thread carlaThread = new Thread(carlaConnection);
-            carlaThread.start();
-        }
-
-        String[] bridgePathArray = bridgePath.split(";");
-
-        String path = bridgePathArray[0];
-        String command = bridgePathArray[1];
-
-        // check the current operating system
-        boolean isWindows = System.getProperty("os.name").toLowerCase().startsWith("windows");
-
-        if (isWindows) {
-            command = "cmd.exe /c start " + command;
-        } else {
-            command = "sh " + command;
-        }
-        // connect carla client
-        while (connectionAttempts-- > 0) {
-            boolean connected = true;
-
+        // Initialize XML-RPC connections
+        if (carlaConfig.carlaSensorLibRPCUrl != null || carlaConfig.carlaActorLibRPCUrl != null) {
+            // Use multi-server manager for separate sensor and actor connections
+            multiXmlRpcManager = new CarlaMultiXmlRpcManager();
+            
             try {
-                connectionProcess = Runtime.getRuntime().exec(command, null, new File(path));
-            } catch (Exception ex) {
-                ex.printStackTrace();
-                if (connectionAttempts == 0) {
-                    log.info("Maximum connection attempts reached and connecting to CARLA simulator failed.");
-                } else {
-                    log.warn("Error while connecting to CARLA simulator. Retrying.");
+                // Add sensor library server
+                // convert string null to null
+                if ("null".equalsIgnoreCase(carlaConfig.carlaSensorLibRPCUrl)) {
```

Comment:
Not sure what is this line and line 307 do, if it is necessary to convert that from string to null, then we will have problem in line 297 since that is raw data without converting